core/src/common/cdk-deployer.ts

```diff
@@ -0,0 +1,359 @@
+            's3:PutBucketAcl',
+            's3:PutEncryptionConfiguration',
+            's3:PutBucketPublicAccessBlock',
+            's3:PutBucketVersioning',
+            's3:DeleteBucket',
+            's3:PutBucketPolicy',
+
+          ],
+        }),
+        new PolicyStatement({
+          resources: [
+            `arn:aws:cloudformation:${Aws.REGION}:${Aws.ACCOUNT_ID}:stack/CDKToolkit*`
+          ],
+          actions: [
+            'cloudformation:DescribeStacks',
+            'cloudformation:DeleteStack',
+            'cloudformation:DeleteChangeSet',
+            'cloudformation:CreateChangeSet',
+            'cloudformation:DescribeChangeSet',
+            'cloudformation:ExecuteChangeSet',
+            'cloudformation:DescribeStackEvents',
+            'cloudformation:GetTemplate',
+          ],
+        }),
+        new PolicyStatement({
+          resources: [
+            cdkDeployRole.roleArn,
+            cdkPublishRole.roleArn,
+          ],
+          actions: [
+            'sts:AssumeRole',
+          ],
+        }),
+        new PolicyStatement({
+          resources: [
+            `arn:aws:ssm:${Aws.REGION}:${Aws.ACCOUNT_ID}:parameter/cdk-bootstrap/*/*`,
+          ],
+          actions: [
+            'ssm:PutParameter',
+            'ssm:GetParameters',
+          ],
+        }),
+        new PolicyStatement({
+          resources: [
+            `arn:aws:ecr:${Aws.REGION}:${Aws.ACCOUNT_ID}:repository/cdk*`,
+          ],
+          actions: [
+            'ecr:SetRepositoryPolicy',
+            'ecr:GetLifecyclePolicy',
+            'ecr:PutImageTagMutability',
+            'ecr:DescribeRepositories',
+            'ecr:ListTagsForResource',
+            'ecr:PutImageScanningConfiguration',
+            'ecr:CreateRepository',
+            'ecr:PutLifecyclePolicy',
+            'ecr:SetRepositoryPolicy',
+            'ecr:DeleteRepository',
+            'ecr:TagResource',
+          ],
+        }),
+        new PolicyStatement({
+          resources: [
+            `arn:aws:iam::${Aws.ACCOUNT_ID}:role/cdk*`,
+          ],
+          actions: [
+            'iam:GetRole',
+            'iam:CreateRole',
+            'iam:TagRole',
+            'iam:DeleteRole',
+            'iam:AttachRolePolicy',
+            'iam:DetachRolePolicy',
+            'iam:GetRolePolicy',
+            'iam:PutRolePolicy',
+            'iam:DeleteRolePolicy',
+          ],
+        }),
+        new PolicyStatement({
+          resources: [
+            `arn:aws:logs:${Aws.REGION}:${Aws.ACCOUNT_ID}:log-group:/aws/codebuild/*`,
+          ],
+          actions: [
+            'logs:PutLogEvents',
+          ],
+        }),
+      ]
+    }))
+
+    const codeBuildProject = new Project(this, 'CodeBuildProject', {
+      source: Source.gitHub({
+        owner: props.githubRepository.split('/')[0],
+        repo: props.githubRepository.split('/')[1],
+        reportBuildStatus: true,
+      }),
+      buildSpec: BuildSpec.fromObject({
```

Comment:
this is overwritten by startBuild custom resource

Reply:
agree we can skip passing a buildspec here